File: java/sql_injection/ProductQueries.java
import java.sql.*;

public class ProductQueries {
    private static final String CONNECTION_URL = "jdbc:postgresql://prod-db:5432/shopist";
    private static final String DB_USER = "app";
    private static final String DB_PASS = "apppassword";

    // VULN 1: String concatenation SQL injection - product search
    public ResultSet searchProducts(String searchTerm) throws SQLException {
        Connection conn = DriverManager.getConnection(CONNECTION_URL, DB_USER, DB_PASS);
        Statement stmt = conn.createStatement();
        String query = "SELECT id, name, price, stock FROM products WHERE name LIKE '%" +
            searchTerm + "%' OR description LIKE '%" + searchTerm + "%'";
        return stmt.executeQuery(query);
    }

    // VULN 2: String.format SQL injection - price range filter
    public ResultSet getProductsByPriceRange(String minPrice, String maxPrice) throws SQLException {
        Connection conn = DriverManager.getConnection(CONNECTION_URL, DB_USER, DB_PASS);
        Statement stmt = conn.createStatement();
        String query = String.format(
            "SELECT * FROM products WHERE price BETWEEN %s AND %s ORDER BY price ASC",
            minPrice, maxPrice
        );
        return stmt.executeQuery(query);
    }

    // VULN 3: String concatenation with ORDER BY injection - category filter
    public ResultSet getProductsByCategory(String category, String sortField) throws SQLException {
        Connection conn = DriverManager.getConnection(CONNECTION_URL, DB_USER, DB_PASS);
        Statement stmt = conn.createStatement();
        String query = "SELECT * FROM products WHERE category = '" + category + "' ORDER BY " + sortField;
        return stmt.executeQuery(query);
    }
}

File: kotlin/sql_injection/ProductQueries.kt
import java.sql.Connection
import java.sql.DriverManager
import java.sql.ResultSet

class ProductQueries {
    private val connectionUrl = "jdbc:postgresql://prod-db:5432/shopist"
    private val dbUser = "app"
    private val dbPass = "apppassword"

    // VULN 1: Kotlin string template SQL injection - product search
    fun searchProducts(searchTerm: String): ResultSet {
        val conn: Connection = DriverManager.getConnection(connectionUrl, dbUser, dbPass)
        val stmt = conn.createStatement()
        val query = "SELECT id, name, description, price FROM products WHERE name LIKE '%$searchTerm%' OR description LIKE '%$searchTerm%'"
        return stmt.executeQuery(query)
    }

    // VULN 2: String concatenation SQL injection - price range query
    fun getProductsByPriceRange(minPrice: String, maxPrice: String): ResultSet {
        val conn: Connection = DriverManager.getConnection(connectionUrl, dbUser, dbPass)
        val stmt = conn.createStatement()
        val query = "SELECT * FROM products WHERE price >= " + minPrice + " AND price <= " + maxPrice + " ORDER BY price ASC"
        return stmt.executeQuery(query)
    }

    // VULN 3: StringBuilder SQL injection - category filter
    fun getProductsByCategory(category: String, sortField: String): ResultSet {
        val conn: Connection = DriverManager.getConnection(connectionUrl, dbUser, dbPass)
        val stmt = conn.createStatement()
        val query = StringBuilder()
            .append("SELECT id, name, price, stock FROM products WHERE category = '")
            .append(category)
            .append("' ORDER BY ")
            .append(sortField)
            .toString()
        return stmt.executeQuery(query)
    }
}
